app/src/main/java/com/x8bit/bitwarden/data/platform/manager/model/SpecialCircumstance.kt

```diff
@@ -97,4 +89,23 @@ sealed class SpecialCircumstance : Parcelable {
      */
     @Parcelize
     data object VaultShortcut : SpecialCircumstance()
+
+    /**
+     * A subset of [SpecialCircumstance] that are only relevant in a pre-login state and should be
+     * cleared after a successful login.
+     *
+     * @see [SpecialCircumstanceManager.clearSpecialCircumstanceAfterLogin]
+     */
+    @Parcelize
+    sealed class PreLogin : SpecialCircumstance() {
```

Comment:
So this totally works, I am curious what you think of this method vs the abstract boolean approach?

Just adding `abstract val shouldClearOnLogin: Boolean` to the class.

Reply:
6 in one for the current use case, I do like the idea of having these types of in their own set/category, but that just works better for how I like to organize. I can appreciate the preference to avoid reliance on inheritance/typing stuff so I won't be hurt if you feel strongly going that route. LMK 👍

Reply:
So this causes an issue with the Parcelable impl on the data objects, they would need be made data classes to be able to parcelize the `shouldClearOnLogin: Boolean`.

Reply:
While I am fine with this implementation, you do not need to do that. You just have to make it a getter.

```kotlin
object class Blah : SpecialCircumstance() {
    override val shouldClearOnLogin: Boolean get() = true
}
```